### PR TITLE
Improve handling constrain violation issue in JDBC drivers

### PR DESCRIPTION
**Pull request by GANGANI**
### Proposed changes in this pull request
$Subject

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java [hunk 1]
```diff
@@ -303,7 +303,9 @@ private void insertAccessToken(String accessToken, String consumerKey, AccessTok
             // Handle constrain violation issue in JDBC drivers which does not throw
             // SQLIntegrityConstraintViolationException or SQLException.
             if (StringUtils.containsIgnoreCase(e.getMessage(), "CON_APP_KEY") || (e.getCause() != null &&
-                    StringUtils.containsIgnoreCase(e.getCause().getMessage(), "CON_APP_KEY"))) {
+                    StringUtils.containsIgnoreCase(e.getCause().getMessage(), "CON_APP_KEY"))
+                    && (e.getCause().getCause() != null &&
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java:307]:
```suggestion
                    || (e.getCause().getCause() != null &&
```

Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java]: Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1780/commits/dcc626644ba7a009d06930b93c9fd63641d9202d

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java [hunk 2]
```diff
@@ -303,7 +303,9 @@ private void insertAccessToken(String accessToken, String consumerKey, AccessTok
             // Handle constrain violation issue in JDBC drivers which does not throw
             // SQLIntegrityConstraintViolationException or SQLException.
             if (StringUtils.containsIgnoreCase(e.getMessage(), "CON_APP_KEY") || (e.getCause() != null &&
-                    StringUtils.containsIgnoreCase(e.getCause().getMessage(), "CON_APP_KEY"))) {
+                    StringUtils.containsIgnoreCase(e.getCause().getMessage(), "CON_APP_KEY"))
+                    || (e.getCause().getCause() != null &&
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java:307]:
```suggestion
                    || (e.getCause() != null && e.getCause().getCause() != null &&
```

Comment (GANGANI) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java]: Fixed

**Review by senthalan — APPROVED** [review]

**Pull request merged**